assets/js/hooks/canvas.js

```diff
@@ -0,0 +1,77 @@
+import { getAttributeOrThrow, parseInteger } from "../lib/attribute";
+import { globalPubSub } from "../lib/pub_sub";
+import "gridstack/dist/gridstack.min.css";
+import { GridStack } from "gridstack";
+
+/**
+ * A hook for creating a canvas with gridstackjs.
+ */
+const Canvas = {
+  mounted() {
+    this.props = this.getProps();
+    console.log("Gridstack mounted");
+    const self = this;
+
+    const options = {
+      styleInHead: true,
+      float: true,
+      resizable: { handles: "all" },
+      margin: 0,
+      cellHeight: "4rem",
+    };
+
+    this.grid = GridStack.init(options, this.el);
+
+    this.handleEvent("reload", ({ payload }) => {
+      const grid_items = Object.entries(payload).map(([id, value]) => ({
+        id,
+        ...value,
+      }));
+      this.grid.load(grid_items);
+      Array.from(this.grid.el.children).forEach((item) => {
+        console.log(item.attributes);
+        const output_id = `[id^=outputs-${item.attributes["gs-id"].value}]`;
+        const output_el = document.querySelector(output_id);
+        item.firstChild.appendChild(output_el);
+      });
```

Comment:
Probably very bad idea to do it this way, but I couldn't come up with another solution...
I just remount the existing outputs into the gridstack items

Reply:
didn't put more effort into it yet, because I'm tinkering around with implementing a liveview compatible drag&drop solution